### PR TITLE
desktop: update docs

### PR DESCRIPTION
**Pull request by nsakaimbo**
### Description

Updates desktop app docs:

- [x] Remove irrelevant doc on desktop secrets
- [x] Update release docs
- [x] Update development docs
- [x] Update main README

### CODE REVIEW

File: desktop/README.md [hunk 1]
```diff
@@ -26,6 +24,13 @@ export DEBUG='*'
 ./desktop/release/mac/WordPress.com.app/Contents/MacOS/WordPress.com
 ```
 
+### Development Mode
+
+For a more quicker development experience, the application can be run in "development mode", which executes the application code directly (i.e. without building the app). This development mode can be ran against production WordPress.com webapp, or a local instance of Calypso.
```
Comment (sandymcfadden) [inline desktop/README.md]: Tiny suggestion. I'd remove `more` from the first sentence here. Leaving it as "For a quicker development experience,"

File: desktop/docs/development.md [hunk 2]
```diff
@@ -15,43 +15,34 @@ We use Electron's [IPC](https://github.com/atom/electron/blob/master/docs/api/ip
...
 
 - [Config](../desktop-config/README.md) - app configuration values
-- [Libraries](../../client/desktop/lib/README.md) - details of the local libraries used
-- [App Handlers](../../client/desktop/app-handlers/README.md) - handlers that run before the main window is created
-- [Window Handlers](../../client/desktop/window-handlers/README.md) - handlers that run after the main window is created
+- [Libraries](../app/lib/README.md) - details of the local libraries used
+- [App Handlers](../app/app-handlers/README.md) - handlers that run before the main window is created
+- [Window Handlers](../app/window-handlers/README.md) - handlers that run after the main window is created
+
+The application can also be run in a development mode that uses the localhost instance of Calypso, instead of loading the production WordPress.com webapp. Refer to the main README for details.
```
Comment (sandymcfadden) [inline desktop/docs/development.md]: We could likely add a link to the main README here to make it easy to get to and match the others above.

**Review by sandymcfadden — APPROVED** [review]
Just a couple of minor suggestions as other comments but other then that it looks good!

**Comment by nsakaimbo** [comment]
Thanks for the review, @sandymcfadden! Addressed your feedback, and made a few more little tweaks.

**Pull request merged**